src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -63,6 +64,69 @@ public void Kill()
             }
         }
 
+        /// <summary>Stops the associated process immediately.</summary>
+        private void Stop()
+        {
+            EnsureState(State.HaveNonExitedId);
+            if (Interop.Sys.Kill(_processId, Interop.Sys.Signals.SIGSTOP) != 0)
+            {
+                throw new Win32Exception(); // to match behavior of Kill()
+            }
+        }
+
+        private IEnumerable<Exception> KillTree()
+        {
+            List<Exception> exceptions = new List<Exception>();
+
+            try
+            {
+                // Stop but don't kill the process. Keeps additional children from being started but leaves the process alive 
+                // so that its children can be enumerated.
+                Stop();
```

Comment:
When Stop returns, the process may be _stopping_ instead of being _stopped_.

Reply:
Thanks! I didn't realize that. Is it only when stopping has completed that we are guaranteed the process can't start child processes? Is there an (easy) way to wait for stopping to complete?

Reply:
From the man page it seems you can do this with `waitid`. I'm not familiar with using that API and its quirks. Maybe you can add a comment about this.

Reply:
Thanks! Comment added.